crates/build/src/lib.rs

```diff
@@ -116,7 +116,7 @@ const VERSION: &str = env!("CARGO_PKG_VERSION");
 pub(crate) mod linting {
     /// Toolchain used to build ink_linting:
     /// https://github.com/paritytech/ink/blob/master/linting/rust-toolchain.toml
-    pub const TOOLCHAIN_VERSION: &str = "nightly-2023-12-28";
+    pub const TOOLCHAIN_VERSION: &str = "nightly-2024-02-07";
```

Comment:
`TOOLCHAIN_VERSION` is used to check that the user has the required toolchain version before compiling and guide them to install the that version. Otherwise, the user will see somewhat confusing error messages from `cargo dylint`.

The version should be exactly the same [as specified in the linter](https://github.com/paritytech/ink/blob/master/linting/rust-toolchain.toml), because it depends on the unstable rustc API.

So, I understand the reason why these changes in the CI are needed, and it makes sense for me.
But can we left the value of `TOOLCHAIN_VERSION` as it is?

Reply:
Done